internal/session/handle.go

```diff
@@ -52,47 +52,46 @@ func (s *Session) handleOther(
 func (s *Session) handleCommand(
 	ctx context.Context,
 	tag string,
-	cmd *proto.Command,
+	cmd command.Payload,
 	ch chan response.Response,
 ) error {
-	switch {
+	switch cmd.(type) {
 	case
-		cmd.GetCapability() != nil,
-		cmd.GetIdGet() != nil,
-		cmd.GetIdSet() != nil,
-		cmd.GetNoop() != nil:
+		*command.Capability,
+		*command.IDGet,
+		*command.IDSet,
+		*command.Noop:
 		return s.handleAnyCommand(ctx, tag, cmd, ch)
 
 	case
-		cmd.GetAuth() != nil,
```

Comment:
What's the current behaviour if an IMAP server _does_ send us the auth command? Will we return NO or BAD? Could we implement some minimal parsing to ensure we can detect AUTHENTICATE and return NO?

Reply:
Right now it's a parser error, will return the error to the client with bad

Reply:
How much work would it be to enable us to return NO here? BAD usually indicates protocol-level errors whereas NO indicates feature-level errors. I'd prefer if we could implement the minimal amount of parsing necessary to be able to reject AUTHENTICATE requests with NO rather than BAD.

Reply:
Have to parse the full Authenticate command to get this recognized properly.